Added enum iteration example (#8599)
godotengine/godot-docs: tutorials/scripting/gdscript/gdscript_basics.rst

    const TILE_SPIKE = 2
    const TILE_TELEPORT = 3

> 

Reviewer: Please restore this, there's no reason to remove it

::

>     enum States {STATE_IDLE, STATE_JUMP = 5, STATE_SHOOT}

Reviewer: Restore the name and the empty line, it matters for readability, and it makes more sense for this to be singular

    # Is the same as:
    const States = {STATE_IDLE = 0, STATE_JUMP = 5, STATE_SHOOT = 6}
    # Access values with States.STATE_IDLE, etc.

    func _ready():
        # Access values with Name.KEY, prints '5'
        print(States.STATE_JUMP)
        # Use constant dictionary functions
        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'
        print(States.keys())
        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'
        print(States)
        # prints '[0, 5, 6]'

>         print(States.values())

Reviewer: ```suggestion
        print(States.values())

```
Restore this whitespace

    # Is the same as:
    const State = {STATE_IDLE = 0, STATE_JUMP = 5, STATE_SHOOT = 6}
    # Access values with State.STATE_IDLE, etc.

    func _ready():
        # Access values with Name.KEY, prints '5'
        print(State.STATE_JUMP)
        # Use constant dictionary functions
        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'
        print(State.keys())
        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'

>         print(State)

Reviewer 2: Wait...

This is not true. `State` is a Dictionary so it should print `{ "STATE_IDLE": 0, "STATE_JUMP": 5, "STATE_SHOOT": 6 }`.

    # Is the same as:
    const State = {STATE_IDLE = 0, STATE_JUMP = 5, STATE_SHOOT = 6}
    # Access values with State.STATE_IDLE, etc.

    func _ready():
        # Access values with Name.KEY, prints '5'
        print(State.STATE_JUMP)
        # Use constant dictionary functions
        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'
        print(State.keys())

> 

Author: Should I restore the whitespace here? If yes, should I place it between the below print statements so they aren't a united codeblock?

Reviewer: Put it after the last line of code, keep two lines before "Functions", easier to read

        # Use constant dictionary functions
        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'
        print(State.keys())
        # prints '[0, 5, 6]'
        print(State.values())

> And since the keys are inside a dictionary, if you wish to iterate over the values assigned to those enums, you can use the :ref:`Dictionary.values() <class_Dictionary_method_values>` method.

Reviewer: I'd say to remove the rest, it's sufficient to have the above

> And since the keys are inside a dictionary, if you wish to iterate over the values assigned to those enums, you can use the :ref:`Dictionary.values() <class_Dictionary_method_values>` method.
> 
> :: 
> 
>     enum States {STATE_IDLE = 0, STATE_JUMP = 5, STATE_SHOOT = 6}
> 
>     func _ready():
>         for picked_state in States:
>             print(picked_state) #This will print: "STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"
> 
>         for picked_state in States.values():
>             print(picked_state) #This will print: 0, 5, 6

Reviewer: ```suggestion
```

        # prints '["STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"]'

>         print(State.keys())

Reviewer: ```suggestion
        print(State.keys())
        # prints '[0, 5, 6]'
        print(State.values())
```
Like so, as per the suggestion above

> And since the keys are inside a dictionary, if you wish to iterate the values assigned to those enums, you must iterate the values of the enums

Reviewer: ```suggestion
And since the keys are inside a dictionary, if you wish to iterate over the values assigned to those enums, you can use the :ref:`Dictionary.values() <class_Dictionary_method_values>` method.
```

:: 

    enum States {STATE_IDLE = 0, STATE_JUMP = 5, STATE_SHOOT = 6}

    func _ready():
        for picked_state in States:

>             print(picked_state)#This will print: "STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"

Reviewer: ```suggestion
            print(picked_state) #This will print: "STATE_IDLE", "STATE_JUMP", "STATE_SHOOT"
```

        for picked_state in States.values():

>             print(picked_state)#This will print: 0, 5, 6

Reviewer: ```suggestion
            print(picked_state) #This will print: 0, 5, 6
```
